src/extension-client/remix-extension.ts

```diff
@@ -16,8 +20,20 @@ export class RemixExtension<T extends Api = any> {
   protected currentRequest: PluginRequest
   public isLoaded = false
 
-  constructor() {
+  constructor(opts: RemixExtensionOptions) {
```

Comment:
Change `opts` to `options` if it's ok. It's more readable for language service tools.
Also make it optional: `constructor(options?: RemixExtensionOptions`.